### PR TITLE
Refactor Event publisher proxy tests

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Refactor Event publisher proxy tests to make it more readable and reusable:

- Use table tests and remove the use of `exec` functions.
- Introduce parallelism in the tests (where possible).
- Remove not needed `nolint` annotations (where possible).
- Rename test files to reflect the scope of the test.
- Move tests under their correct packages.

### CODE REVIEW

**Review by nachtmaar — COMMENTED** [review]
Thanks for four PR 😎
Having a short summary of what the PR is about would be nice 😃 I think in this case it is about (1) getting rid of the exec-style table-driven tests and (2) parallelising the test, right ?

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+			// nolint:scopelint
+			for _, testCase := range innerTestCases {
+				t.Run(testCase.name, func(t *testing.T) {
+					body, headers := testCase.provideMessage()
+					resp, err := testingutils.SendEvent(publishLegacyEndpoint, body, headers)
+					require.NoError(t, err)
+					require.Equal(t, testCase.wantStatusCode, resp.StatusCode)
+					if testCase.givenEndpoint == publishLegacyEndpoint {
+						handlertest.ValidateErrorResponse(t, *resp, &testCase.wantResponse)
+					}
```
[the remark this answers is not in the document]
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go:255]: rough opinion is that team prefers one liner funcs when there is only one statement in. So exactly as you did 👍

File: components/event-publisher-proxy/pkg/handler/beb/mock/mock.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+	client := oauth.NewClient(ctx, mock.cfg)
+	defer client.CloseIdleConnections()
+
+	msgReceiver := receiver.NewHTTPMessageReceiver(mock.cfg.Port)
+	msgSender := sender.NewBebMessageSender(mock.cfg.EmsPublishURL, client)
+	msgHandlerOpts := &options.Options{MaxRequestSize: int64(requestSize)}
+	msgHandler := beb.NewHandler(msgReceiver, msgSender, mock.cfg.RequestTimeout, mock.legacyTransformer, msgHandlerOpts, mock.subscribedProcessor, mock.logger, mock.collector)
+
+	go func() { require.NoError(t, msgHandler.Start(ctx)) }()
```
[the remark this answers is not in the document]
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/handler/beb/mock/mock.go:127]: rough opinion is that team prefers one liner funcs when there is only one statement in. So exactly as you did 👍

File: components/event-publisher-proxy/pkg/handler/beb/mock/mock.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+// GetReadinessEndpoint returns the readiness endpoint used by the BebHandlerMock.
+func (m *BebHandlerMock) GetReadinessEndpoint() string {
+	return m.readinessEndpoint
+}
+
+// StartOrDie starts a new BebHandlerMock instance or die if a precondition fails.
+// Preconditions: 1) beb.Handler started without errors.
+func StartOrDie(ctx context.Context, t *testing.T, requestSize int, eventTypePrefix, eventsEndpoint string, // nolint:unparam
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/mock/mock.go]: Removed.

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+			for _, testCase := range handlertest.TestCasesForLegacyEvents {
+				t.Run(testCase.Name, func(t *testing.T) {
+					body, headers := testCase.ProvideMessage()
+					resp, err := testingutils.SendEvent(publishLegacyEndpoint, body, headers)
+					require.NoError(t, err)
+					require.Equal(t, testCase.WantStatusCode, resp.StatusCode)
+					if testCase.WantStatusCode == http.StatusOK {
+						handlertest.ValidateOkResponse(t, *resp, &testCase.WantResponse)
+					} else {
+						handlertest.ValidateErrorResponse(t, *resp, &testCase.WantResponse)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go]: Done.

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+					provideMessage: func() (string, http.Header) {
+						return fmt.Sprintf(`"%s"`, testingutils.CloudEventData), testingutils.GetBinaryMessageHeaders()
+					},
+					givenEndpoint:  publishEndpoint,
+					wantStatusCode: http.StatusBadRequest,
+				},
+			}
+			// nolint:scopelint
+			for _, testCase := range innerTestCases {
+				t.Run(testCase.name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go:250]: As discussed, this is not needed anymore.

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go]: I agree. Let's do this in another ticket.

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go [hunk 6]
```diff
@@ -0,0 +1,502 @@
...
+
+			handlerMock := mock.StartOrDie(context.TODO(), t, requestSize, tc.givenEventTypePrefix, eventsEndpoint, requestTimeout, serverResponseTime,
+				mock.WithApplication(tc.givenApplicationNameToCreate, tc.givenApplicationNameToValidate),
+			)
+			defer handlerMock.Close()
+			publishEndpoint := fmt.Sprintf(publishEndpointFormat, handlerMock.GetPort())
+
+			// nolint:scopelint
+			for _, testCase := range handlertest.TestCasesForCloudEvents {
+				t.Run(testCase.Name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go:95]: Done.

File: components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go [hunk 7]
```diff
@@ -0,0 +1,502 @@
...
+		},
+		{
+			name:                           "With empty prefix and not-clean application name",
+			givenEventTypePrefix:           testingutils.MessagingEventTypePrefixEmpty,
+			givenApplicationNameToCreate:   testingutils.ApplicationNameNotClean,
+			givenApplicationNameToValidate: testingutils.ApplicationNameNotClean,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/beb/handler_integration_test.go:77]: Done.

**Review by nachtmaar — APPROVED** [review]

**Pull request merged**